presto-main/src/test/java/io/prestosql/operator/aggregation/TestAccumulatorCompiler.java

```diff
@@ -0,0 +1,88 @@
+import io.prestosql.spi.type.LongTimestamp;
+import io.prestosql.spi.type.RealType;
+import io.prestosql.spi.type.TimestampType;
+import io.prestosql.testng.services.ReportUnannotatedMethods;
+import org.testng.annotations.Test;
+
+import java.lang.invoke.MethodHandle;
+import java.util.Optional;
+
+import static io.prestosql.operator.aggregation.AggregationMetadata.ParameterMetadata.ParameterType.INPUT_CHANNEL;
+import static io.prestosql.operator.aggregation.AggregationMetadata.ParameterMetadata.ParameterType.STATE;
+import static io.prestosql.util.Reflection.methodHandle;
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class TestAccumulatorCompiler
+{
+    @Test
+    public void testAccumulatorCompilerForTypeSpecificObjectParameter()
+    {
+        DynamicClassLoader classLoader = new DynamicClassLoader(TestAccumulatorCompiler.class.getClassLoader());
+
+        TimestampType parameterType = TimestampType.TIMESTAMP_NANOS;
+        assertThat(parameterType.getJavaType()).isEqualTo(LongTimestamp.class);
+
+        Class<? extends AccumulatorState> stateInterface = TestAggregationState.class;
+        AccumulatorStateSerializer<?> stateSerializer = StateCompiler.generateStateSerializer(stateInterface, classLoader);
+        AccumulatorStateFactory<?> stateFactory = StateCompiler.generateStateFactory(stateInterface, classLoader);
+
+        MethodHandle inputFunction = methodHandle(TestAccumulatorCompiler.class, "longTimestampAggregationInput", TestAggregationState.class, LongTimestamp.class);
+        MethodHandle combineFunction = methodHandle(TestAccumulatorCompiler.class, "longTimestampAggregationCombine", TestAggregationState.class, TestAggregationState.class);
+        MethodHandle outputFunction = methodHandle(TestAccumulatorCompiler.class, "longTimestampAggregationOutput", TestAggregationState.class, BlockBuilder.class);
+        AggregationMetadata metadata = new AggregationMetadata(
+                "longTimestampAggregation",
+                ImmutableList.of(
+                        new AggregationMetadata.ParameterMetadata(STATE),
+                        new AggregationMetadata.ParameterMetadata(INPUT_CHANNEL, parameterType)),
+                inputFunction,
+                Optional.empty(),
+                combineFunction,
+                outputFunction,
+                ImmutableList.of(new AggregationMetadata.AccumulatorStateDescriptor(
+                        stateInterface,
+                        stateSerializer,
+                        stateFactory)),
+                RealType.REAL);
+
+        // test if we can compile aggregation
+        assertThat(AccumulatorCompiler.generateAccumulatorFactoryBinder(metadata, classLoader)).isNotNull();
+
+        // TODO test if aggregation actually works...
+    }
+
+    public interface TestAggregationState
+            extends AccumulatorState
+    {
+    }
+
+    @ReportUnannotatedMethods.Suppress
```

Comment:
Just move those static methods into inner class (just as if you were declaring aggregation)